src/v/cluster/topics_frontend.cc

```diff
@@ -81,6 +81,20 @@ ss::future<std::vector<topic_result>> topics_frontend::create_topics(
   std::vector<custom_assignable_topic_configuration> topics,
   model::timeout_clock::time_point timeout) {
     vlog(clusterlog.info, "Create topics {}", topics);
+
+    if (!_features.local().is_active(features::feature::cloud_retention)) {
```

Comment:
we need to make sure that this will be documented

Reply:
I've realised that's not needed. The topic creation request is never forwarded, which means that we don't have to evolve the ADL encoding for it.